autosklearn/pipeline/components/data_preprocessing/categorical_encoding/no_encoding.py

```diff
@@ -1,23 +1,30 @@
+from typing import Dict, Optional, Tuple, Union
+import numpy as np
+
 from ConfigSpace.configuration_space import ConfigurationSpace
+
+from autosklearn.pipeline.base import DATASET_PROPERTIES_TYPE, PIPELINE_DATA_DTYPE
 from autosklearn.pipeline.components.base import \
     AutoSklearnPreprocessingAlgorithm
 from autosklearn.pipeline.constants import DENSE, SPARSE, UNSIGNED_DATA, INPUT
 
 
 class NoEncoding(AutoSklearnPreprocessingAlgorithm):
-    def __init__(self, random_state=None):
+    def __init__(self, random_state: Optional[np.random.RandomState] = None):
```

Comment:
Just to state it here as we discussed offline I created issue https://github.com/automl/auto-sklearn/issues/1150 so that we remove the no encoder when we move to the new scikit learn version.